python/tank/authentication/sso_saml2/core/utils.py

```diff
@@ -154,23 +178,43 @@ def get_saml_claims_expiration(encoded_cookies):
     """
     # Shotgun appends the unique numerical ID of the user to the cookie name:
     # ex: shotgun_sso_session_expiration_u78
-    saml_claims_expiration = _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_expiration_u")
+    saml_claims_expiration = (
+        _get_cookie(encoded_cookies, "shotgun_current_user_sso_claims_expiration") or
+        _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_expiration_u")
+    )
     if saml_claims_expiration is not None:
         saml_claims_expiration = int(saml_claims_expiration)
     return saml_claims_expiration
 
 
-def get_saml_user_name(encoded_cookies):
+def get_session_expiration(encoded_cookies):
+    """
+    Obtain the expiration time of the Shotgun session from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: An int with the time in seconds since January 1st 1970 UTC, or None
```

Comment:
I can see in the code that it returns None because there is no experitation date, but the docstring should explain why so I don't have to look at the code.